common/softwarecontainer-common.cpp

```diff
@@ -75,22 +77,38 @@ bool isDirectoryEmpty(const std::string &path) {
 
 bool isFile(const std::string &path)
 {
-    return S_ISREG(getStat(path).st_mode);
+    struct stat st;
+    if (getStat(path, st)) {
+        return S_ISREG(st.st_mode);
```

Comment:
I might be nitpicking, but maybe `return getStat(path, st) && S_ISREG(st.st_mode);`

I'm OK w/ the semantics here though.